### DISCUSSION TITLE
Optimizing multiple loss functions for similar problems

### DISCUSSION THREAD

**Discussion by marijanarra** [lululxvi/deepxde#1067] [Ideas]
Hi Lu, 

I would like to implement a solution that allows learning the solution of different (but similar enough) PDEs on the same domain or a single PDE on different (but similar enough) domains. The transfer learning approach doesn't reach a good level of generalization so I'd like to try something alike multiple-objective learning. 

At the moment, I'm trying to figure out how to do this using deepXDE, however, the library is seems to be specialized for single domain/PDE problems. I'm wondering if it's worth the effort of restructuring the source code or should I just implement the whole solution in Tensorflow/PyTorch directly.

I would appreciate any input on this idea. 

Thanks,
jerymane

**Comment by lululxvi** [comment]
If you can refine your idea and be specific, I can then answer your question.

**Reply by marijanarra** [reply]
Say I have a bunch of slightly different geometries like these ones:

![polys](https://user-images.githubusercontent.com/64425956/207037013-cb8073f8-78ed-4988-8507-d644eaa3dce5.png)

Say I want to solve a simple Laplace equation on these geometries with BCs dependent on the geometry like defined below. So the PDE solutions are slightly different for each geometry
```
Neumann BCs:
1/L on red edge (L is the red edge length)
1/S on blue edges (S is the geometry area)
```

My goal is to have a single trained PINN model that can output the correct PDE solution, respecting the corresponding BCs for any shape from the training set or similar (I would expect to reach a certain level of generalization with a large enough set of training geometries). 

Currently, PINNs are minimizing a loss function on a set of training points representing a single geometry. My goal would be implement a solution to train a single PINN on multiple geometries simultaneously.

Let me know if you'd need further clarification.

Thanks!

**Reply by lululxvi** [reply]
I am not sure how to do it.
